README.md

```diff
@@ -1,53 +1,290 @@
+services:
+  db:
+    image: mariadb
+    restart: always
+    volumes:
+      - db:/var/lib/mysql
+    environment:
+      - MYSQL_ROOT_PW=...
+      - MYSQL_USER_PW=...
+      - MYSQL_DATABASE=nextcloud
+      - MYSQL_USER=nextcloud
 
-## ... via [`docker-compose`](https://github.com/docker/compose)
+  app:
+    image: nextcloud
+    links:
+      - db
+    volumes:
+      - data:/var/www/html/data
+      - config:/var/www/html/config
+      - apps:/var/www/html/custom_apps
+    restart: always
 
-The recommended minimal setup is using this image in combination with two containers: A database container and a reverse proxy for the http connection to the service.
-A working example can be found at [IndieHosters/Nextcloud](https://github.com/indiehosters/nextcloud).
+  web:
+    image: nginx
+    ports:
+      - "80:80"
+    links:
+      - app
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+    volumes-from:
+      - app
+    restart: always
+```
+
+## First use
+When you first access your nextcloud, the setup wizard will appear and ask you to choose an administrator account, password and the database connection. For the database use `db` as host and `nextcloud` as table and user name. Also enter the password you chose in the compose file.
+
+## SSL encryption
+Until here, we haven't talked about encrypting the connection between your nextcloud host and the clients. Using up-to-date encryption is mandatory if your host is reachable from the internet. There are many different possibilities to introduce encryption. 
 
-If you want to access your Nextcloud from the internet we recommend configuring your reverse proxy to use encryption (for example via [let's Encrypt](https://letsencrypt.org/))
+An easy and free way to get certificates that are accepted by the browsers is [Let's Encrypt](https://letsencrypt.org/). The whole certificate generation / validation is fully automated and certificate renewals are also very easy. 
+To integrate Let's Encrypt, we recommend using a reverse proxy in front of our nextcloud installation. Your nextcloud will only be reachable through the proxy, which encrypts all traffic to the clients. See our [examples](https://github.com/nextcloud/docker/tree/master/.examples) to get an idea how it works.
 
-## Update to a newer version
 
-To update your Nextcloud version you simply have to pull and start the new container.
+# Update to a newer version
+Updating the nextcloud container is done by pulling the new image and throwing away the old container. Since all data is stored in volumes nothing gets lost. The startup script will check for the version in your data and the installed docker version. If it finds a mismatch, it automatically starts the upgrade process. 
+
 ```console
 $ docker pull nextcloud
 $ docker run -d nextcloud
 ```
-When you access your site the update wizard will show up.
+
+When using docker-compose:
+
+```console
+$ docker-compose up -d --pull
```

Comment:
is can't find the `--pull` option in the docs https://docs.docker.com/compose/reference/up/